pkgs/development/libraries/libedit/default.nix

```diff
@@ -21,9 +21,9 @@ stdenv.mkDerivation rec {
   # There is a DR to fix this issue with Clang which is not merged
   # yet.
   # https://reviews.llvm.org/D137043
-  NIX_CFLAGS_COMPILE = lib.optional
-    (stdenv.targetPlatform.isMusl && stdenv.cc.isClang)
-    "-D__STDC_ISO_10646__=201103L";
+  env.NIX_CFLAGS_COMPILE =
+    if (stdenv.targetPlatform.isMusl && stdenv.cc.isClang)
+    then "-D__STDC_ISO_10646__=201103L" else "";
```

Comment:
```suggestion
    lib.optionalString (stdenv.targetPlatform.isMusl && stdenv.cc.isClang)
      "-D__STDC_ISO_10646__=201103L";
```

Reply:
That's the function I was looking for, thank you!